### PR TITLE
Simplify some Option / Result / ? operator patterns

### PR DESCRIPTION
**Pull request by huitseeker**
When they identically match a combinator on those types.

Tool-aided by [comby-rust](https://github.com/huitseeker/comby-rust).

### CODE REVIEW

File: node/network/statement-distribution/src/lib.rs [hunk 1]
```diff
@@ -1310,10 +1310,7 @@ async fn handle_incoming_message<'a>(
 		metrics,
 	).await;
```
Comment (bkchr) [inline node/network/statement-distribution/src/lib.rs:1311]:
```suggestion
	).await?;
```

File: node/network/statement-distribution/src/lib.rs [hunk 2]
```diff
@@ -1310,10 +1310,7 @@ async fn handle_incoming_message<'a>(
...
 
-	let statement = match statement {
-		None => return None,
-		Some(statement) => statement,
-	};
+	let statement = statement?;
```
Comment (bkchr) [inline node/network/statement-distribution/src/lib.rs:1313]:
```suggestion
```

File: bridges/relays/ethereum/src/main.rs [hunk 3]
```diff
@@ -270,14 +270,10 @@ fn ethereum_deploy_contract_params(matches: &clap::ArgMatches) -> Result<Ethereu
 }
 
 fn ethereum_exchange_submit_params(matches: &clap::ArgMatches) -> Result<EthereumExchangeSubmitParams, String> {
-	let eth_nonce = if let Some(eth_nonce) = matches.value_of("eth-nonce") {
-		Some(
+	let eth_nonce = matches.value_of("eth-nonce").map(|eth_nonce| {
```
Comment (bkchr) [inline bridges/relays/ethereum/src/main.rs:273]:
```suggestion
	let eth_nonce = matches.value_of("eth-nonce").map(|eth_nonce|
```

File: bridges/relays/ethereum/src/main.rs [hunk 4]
```diff
@@ -270,14 +270,10 @@ fn ethereum_deploy_contract_params(matches: &clap::ArgMatches) -> Result<Ethereu
...
-	let eth_nonce = if let Some(eth_nonce) = matches.value_of("eth-nonce") {
-		Some(
+	let eth_nonce = matches.value_of("eth-nonce").map(|eth_nonce| { 
 			relay_ethereum_client::types::U256::from_dec_str(&eth_nonce)
 				.map_err(|e| format!("Failed to parse eth-nonce: {}", e))?,
-		)
-	} else {
-		None
-	};
+		 });
```
Comment (bkchr) [inline bridges/relays/ethereum/src/main.rs:276]:
```suggestion
		);
```

**Review by bkchr — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**